src/server/notifications/WebHookSubscription2021/WebHookWebId.ts

```diff
@@ -0,0 +1,31 @@
+import { OkResponseDescription } from '../../../http/output/response/OkResponseDescription';
+import type { ResponseDescription } from '../../../http/output/response/ResponseDescription';
+import { BasicRepresentation } from '../../../http/representation/BasicRepresentation';
+import { TEXT_TURTLE } from '../../../util/ContentTypes';
+import { trimTrailingSlashes } from '../../../util/PathUtil';
+import type { OperationHttpHandlerInput } from '../../OperationHttpHandler';
+import { OperationHttpHandler } from '../../OperationHttpHandler';
+
+/**
+ * The WebHookSubscription2021 requires the server to have a WebID
+ * that is used during the generation of the DPoP headers.
+ * There are no real specifications about what this should contain or look like,
+ * so we just return a turtle document that contains a solid:oidcIssuer triple for now.
+ * This way we confirm that our server was allowed to sign the token.
+ */
+export class WebHookWebId extends OperationHttpHandler {
+  private readonly issuer: string;
+
+  public constructor(baseUrl: string) {
+    super();
+    this.issuer = trimTrailingSlashes(baseUrl);
+  }
+
+  public async handle(input: OperationHttpHandlerInput): Promise<ResponseDescription> {
+    const webId = `@prefix solid: <http://www.w3.org/ns/solid/terms#>.
+<> solid:oidcIssuer <${this.issuer}>.`;
```

Comment:
If we want to generate Turtle this way, we'll need strong validation on `issuer`.
Let's see if it passes the `URL` parser.

Reply:
Although we could also just use relative URLs.

Reply:
Since this block is fixed I could run it through the N3 parser in the constructor as a check. Potentially with a dummy base URL if that is required.

Reply:
Let's use the N3 serializer then to create it 😃

Reply:
Then we can't do it in the constructor though (which is where I think we want to throw the error if there is an issue with the issuer) since we don't have the value for the subject (which is why we use `<>`). So planning to still just use the serializer instead.

Reply:
Right! Whatever is more simple indeed.